Initial nightly job configurations

DanielXiao opened the pull request:
Use Jenkins job builder to generate job configurations for
nightly test jobs.

DanielXiao committed:
Initial nightly job configurations

Use Jenkins job builder to generate job configurations for
nightly test jobs.

zjs commented on jenkins/jjb/project.yaml:
```diff
@@ -0,0 +1,29 @@
+- project:
+    name: 'nightly-tests-trigger'
```
Let's use `scenario` instead of `nightly` in these files (as you have already done in the file names) to be consistent with the current terminology.

DanielXiao replied:
OK. I will use scenario.

DanielXiao replied:
How about change file names to nightly? Scripts in other repos are named by nightly.

zjs replied:
The goal is to eventually replace the term "nightly" with "scenario". Changing everything all-at-once is hard, but if we use "scenario" in new code and when we're updating something for other reasons, we'll slowly switch over.

zjs commented on jenkins/jjb/project.yaml:
```diff
@@ -0,0 +1,29 @@
+          repo: 'vic'
+          version: '1.4.3'
+          branch: 'releases/1.4.3'
+          addition_env: 'VCH_BRANCH=releases/1.4.3'
```
What's the purpose of this?

DanielXiao replied:
To generate similar jobs from the same template.       addition_env: 'VCH_BRANCH=releases/1.4.3' is a workaround because I can't set VCH_BRANCH=master.

DanielXiao replied:
There is no master folder and build are stored directly under bucket if it is from master branch

zjs replied:
I see. Makes sense!

hickeng commented on jenkins/jjb/templates/nightly-tests.yaml:
```diff
@@ -0,0 +1,116 @@
+- job-template:
```
I suggest changing the filename of this as per @zjs earlier comment - we're trying to move to `scenario` as we expect and want these tests to be usable from github PR triggers eventually.
The nightly tests are a specific _use_ of the scenario tests.

DanielXiao replied:
OK. I will refactor nightly to scenario

hickeng commented on jenkins/jjb/templates/nightly-tests.yaml:
```diff
@@ -0,0 +1,116 @@
+    scm:
+      - git:
+          url: 'git@gitlab.eng.vmware.com:core-build/vic-internal.git'
+          credentials-id: '89094db1-3393-4148-bf08-2363af653ec1'
```
Is there any way to use a non-UUID identifier for the credentials?

This is one of the problems I had went importing/exporting jenkins jobs for Bonneville that moved us towards drone. When we needed to redeploy jenkins we could import the old job specs but the credentials (and other uuid based links) would be broken

DanielXiao replied:
No. So far I didn't find no-UUID identifier for the credentials.

zjs replied:
I think you can change the ID here: https://vic-jenkins.eng.vmware.com/credentials/store/system/domain/_/credential/89094db1-3393-4148-bf08-2363af653ec1/update

DanielXiao replied:
No. I can not update ID after it 's generated. I can create a new credential with meaningful ID but I don't know Passphrase of this private key.

zjs replied:
I'll message you the password for the private key.

hickeng commented on jenkins/jjb/templates/nightly-tests.yaml:
```diff
@@ -0,0 +1,116 @@
+              summary.appendText("<li><b>vCenter</b> - ${{vc_build}}</li>", false)
+              summary.appendText("<li><b>ESXi</b> - ${{esx_build}}</li>", false)
+      - email-ext:
+          recipients: vic-eng@vmware.com
```
I found it quite useful to be able to specify only my own email when running scenario tests against a PR. It also allowed me to avoid spamming the entire group when testing build definition changes.

DanielXiao replied:
This template will not be used for PR test. I created a default template and we can overwrite email-ext section in PR job template.

DanielXiao commented:
I modified them according to your comments

DanielXiao committed:
Add vic-product scenario jobs

Add trigger jobs for  vic-product scenario tests on master
and 1.4.3 branches.
Add scenario tests jobs  for  vic-product on master and
1.4.3 branches.

DanielXiao committed:
Add vic-product scenario jobs

Add trigger jobs for  vic-product scenario tests on master
and 1.4.3 branches.
Add scenario tests jobs  for  vic-product on master and
1.4.3 branches.

hickeng commented on jenkins/jjb/templates/scenario-tests.yaml:
```diff
@@ -0,0 +1,119 @@
+    scm:
+      - git:
+          url: 'git@gitlab.eng.vmware.com:core-build/vic-internal.git'
+          credentials-id: '89094db1-3393-4148-bf08-2363af653ec1'
```
This credential ID should be _specified_ explicitly so that we don't need submit changes to this file if recreating or redeploying jenkins. As the jenkins help notes:
> An internal unique ID by which these credentials are identified from jobs and other configuration. Normally left blank, in which case an ID will be generated, which is fine for jobs created using visual forms. Useful to specify explicitly when using credentials from scripted configuration.

DanielXiao committed:
Update gitlab key

Replace gitlab key to another one with meaningful name ID

DanielXiao commented:
I created https://github.com/vmware/vic-tools/pull/8 to replace this PR.
